gap/Forms.gi

```diff
@@ -1,56 +1,77 @@
         newForm := BilinearFormByMatrix(gramMatrix, field);
     else
-        if HasInvariantSesquilinearForm(group) then
-            gapForm := HermitianFormByMatrix(InvariantSesquilinearForm(group).matrix,
-                                             field);
-        else    
-            formMatrix := UnitaryForm(group);
-            if formMatrix = fail then
-                Error("No preserved unitary form found for <group>");
-            fi;
-            gapForm := HermitianFormByMatrix(formMatrix, field);
+        formMatrix := UnitaryForm(group);
+        if formMatrix = fail then
+            ErrorNoReturn("No preserved unitary form found for <group>");
         fi;
+        gapForm := HermitianFormByMatrix(formMatrix, field);
         newForm := HermitianFormByMatrix(gramMatrix, field);
     fi;
-    # the following if condition can only ever be fulfilled if <group> is an
-    # orthogonal group; there the case of even dimension is problematic since,
-    # in that case, there are two similarity classes of bilinear forms
-    if not WittIndex(gapForm) = WittIndex(newForm) then
-       ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
-                     "described by the Gram matrix <gramMatrix>.");
-    fi;
-    gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
-                                             field);
-    canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
-                                             field);
-    result := MatrixGroup(field, canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
-    if HasSize(group) then
-        SetSize(result, Size(group));
+    # The Form package has a bug for d = 1 so we need to make this exception
+    if not d = 1 then
+        # the following if condition can only ever be fulfilled if <group> is an
+        # orthogonal group; there the case of even dimension is problematic since,
+        # in that case, there are two similarity classes of bilinear forms
+        if not WittIndex(gapForm) = WittIndex(newForm) then
+            ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
+                          "described by the Gram matrix <gramMatrix>.");
+        fi;
+        gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
+                                                 field);
+        canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
+                                                 field);
+        result := MatrixGroup(field, canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
+    
+        # Set useful attributes
+        if HasSize(group) then
+            SetSize(result, Size(group));
+        fi;
+    else
+        # replaces the Witt index check above
+        if (gramMatrix[1, 1] = 0 and formMatrix[1, 1] <> 0) 
+           or (gramMatrix[1, 1] <> 0 and formMatrix[1, 1] = 0) then
```

Comment:
Are these matrices really using integer entries? That's weird... But if not, then comparing to `0` will always yield that they are unequal.

You can also simplify the code further:
```suggestion
        if IsZero(gramMatrix[1, 1]) <> IsZero(formMatrix[1, 1]) then
```
or even
```suggestion
        if IsZero(gramMatrix) <> IsZero(formMatrix) then
```